### PR TITLE
feat(branch-protection): Push/Reviewer actors can be specified by name 

### PR DESCRIPTION
**Pull request by dion-gionet**
Changes the github_branch_protection resource.

This enables us to use the branch protection resource without using a data block just to retrieve the node ID for users/teams. Node IDs can still be used as an input but you can now use a prefix to define a username or team name

The / prefix defines a user (Ex.: /testuser), the orgname/ prefix defines a team. (Ex.: testorg/team)

### CODE REVIEW

**Comment by dion-gionet** [comment]
@kfcampbell Do you have any feedback on this?

File: github/util_v4_branch_protection.go [hunk 1]
```diff
@@ -223,60 +232,89 @@ func branchProtectionResourceData(d *schema.ResourceData, meta interface{}) (Bra
...
+	data, _ := branchProtectionResourceData(d, meta)
+	orgName := meta.(*Owner).name
+l:
 	for _, a := range actors {
-		if a.Actor.Team != (Actor{}) {
-			bypassActors = append(bypassActors, a.Actor.Team.ID.(string))
+		for _, v := range data.BypassPullRequestActorIDs {
+			if (a.Actor.Team.ID != nil && a.Actor.Team.ID.(string) == v) || (a.Actor.User.ID != nil && a.Actor.User.ID.(string) == v) {
+				bypassActors = append(bypassActors, v)
+				continue l
```
Comment (kfcampbell) [inline github/util_v4_branch_protection.go:267]: Is there another way to do this without the continue to the row label? I don't love this approach.

File: github/util_v4_branch_protection.go [hunk 2]
```diff
@@ -223,60 +232,89 @@ func branchProtectionResourceData(d *schema.ResourceData, meta interface{}) (Bra
...
 	}
-
-	return pushActors
+	return dismissalActors
 }
 
-func setBypassPullRequestActorIDs(actors []BypassPullRequestActorTypes) []string {
+func setBypassPullRequestActorIDs(actors []BypassPullRequestActorTypes, d *schema.ResourceData, meta interface{}) []string {
 	bypassActors := make([]string, 0, len(actors))
+	data, _ := branchProtectionResourceData(d, meta)
```
Comment (kfcampbell) [inline github/util_v4_branch_protection.go:260]: Why aren't we handling the error here?

File: github/util_v4_branch_protection.go [hunk 3]
```diff
@@ -223,60 +232,89 @@ func branchProtectionResourceData(d *schema.ResourceData, meta interface{}) (Bra
 	return data, nil
 }
 
-func setDismissalActorIDs(actors []DismissalActorTypes) []string {
-	pushActors := make([]string, 0, len(actors))
+func setDismissalActorIDs(actors []DismissalActorTypes, d *schema.ResourceData, meta interface{}) []string {
+	dismissalActors := make([]string, 0, len(actors))
+	data, _ := branchProtectionResourceData(d, meta)
```
Comment (kfcampbell) [inline github/util_v4_branch_protection.go:237]: Why aren't we handling the error here?

File: github/util_v4_branch_protection.go [hunk 4]
```diff
@@ -223,60 +232,89 @@ func branchProtectionResourceData(d *schema.ResourceData, meta interface{}) (Bra
...
-			bypassActors = append(bypassActors, a.Actor.Team.ID.(string))
+		for _, v := range data.BypassPullRequestActorIDs {
+			if (a.Actor.Team.ID != nil && a.Actor.Team.ID.(string) == v) || (a.Actor.User.ID != nil && a.Actor.User.ID.(string) == v) {
+				bypassActors = append(bypassActors, v)
+				continue l
+			}
 		}
-		if a.Actor.User != (Actor{}) {
-			bypassActors = append(bypassActors, a.Actor.User.ID.(string))
+		if a.Actor.Team.Slug != "" {
```
Comment (kfcampbell) [inline github/util_v4_branch_protection.go:270]: Is this a breaking change for folks who weren't using the newly-introduced `/` character functionality?

File: github/util_v4_branch_protection.go [hunk 5]
```diff
@@ -366,3 +404,71 @@ func getBranchProtectionID(repoID githubv4.ID, pattern string, meta interface{})
...
+	}
+
+	return actors, nil
+}
+
+// Given a string that is either a username or team slug, return the
+// node id of the user or team it is referring to. Team slugs must be provided
+// with the organization name as prefix (Ex.: exampleorg/exampleteam). Usernames
+// must be provided with the "/" prefix otherwise getNodeIDv4 assumes that
+// the provided string is a node ID.
```
Comment (kfcampbell) [inline github/util_v4_branch_protection.go:426]: Is this specification from the GitHub API/documented anywhere public?

File: github/util_v4_branch_protection.go (hunk 3 above)
Comment (dion-gionet) [reply github/util_v4_branch_protection.go]: Added error handling

File: github/util_v4_branch_protection.go (hunk 1 above)
Comment (dion-gionet) [reply github/util_v4_branch_protection.go]: I've changed the checks and removed the labels, is this method acceptable?

**Review by kfcampbell — APPROVED** [review]
Thanks for updating this! Integration tests for this branch protection resource are broken on the main branch and failing here as well with this error:

```
    testing.go:705: Step 0 error: errors during apply:
        
        Error: Could not resolve to a node with the global id of '1234'
        
          on /tmp/tf-test2340247973/main.tf line 8:
          (source code not available)
```

So I don't think this is a regression. Thank you for the contribution.

File: github/util_v4_branch_protection.go (hunk 1 above)
Comment (kfcampbell) [reply github/util_v4_branch_protection.go]: Yep, thank you!

File: github/util_v4_branch_protection.go (hunk 4 above)
Comment (kfcampbell) [reply github/util_v4_branch_protection.go]: Got it, thanks!

File: github/util_v4_branch_protection.go (hunk 5 above)
Comment (kfcampbell) [reply github/util_v4_branch_protection.go:455]: I was thinking more of the `/` prefix vs. infix behavior. Regardless, I think it's intuitive enough.

**Comment by kfcampbell** [comment]
@dion-gionet Are you okay if I merge/release this PR today?

**Pull request merged**